scrapy/spiders/__init__.py

```diff
@@ -23,6 +23,13 @@ class Spider(object_ref):
     name = None
     custom_settings = None
 
+    def __setattr__(self, attr, value):
+        if attr == 'start_url' and not getattr(self, 'start_urls', []):
```

Comment:
Won’t this behave differently depending on the order in which `start_url` and `start_urls` appear in the `Spider` subclass?

Reply:
You are right. I updated it in any case to follow the previous approach, do the check in start_requests.